### ISSUE TITLE
Unremovable dead container

### ISSUE THREAD

**Issue by creynders** [moby/moby#32787]
**Description**

My CI pipeline suddenly broke last week, at the post-mortem I discovered a situation I haven't encountered before: I have a container which shows up "dead" in `docker ps -a`, but when I try to `rm` it, it fails with 

```
Error response from daemon: Driver aufs failed to remove root filesystem 99c1936d84276a2682bba31a4d30fc6618bb984be36fd560b29861864b595f23: no such file or directory
```

And sure enough, there's no `containers/99c1936d84276a2682bba31a4d30fc6618bb984be36fd560b29861864b595f23` directory.

Unfortunately this error won't even allow me to rebuild the container.

**Steps to reproduce the issue:**

Not sure, basically my pipeline builds and ups a container (with docker-compose) and once it's finished doing its stuff, removes the container. This has worked without any troubles before, but suddenly - for no apparent reason - it broke down.

**Output of `docker version`:**

```
Client:
 Version:      1.13.0
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 09:50:17 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 09:50:17 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 37
 Running: 22
 Paused: 0
 Stopped: 15
Images: 195
Server Version: 1.13.0
Storage Driver: aufs
 Root Dir: /data/docker/aufs
 Backing Filesystem: extfs
 Dirs: 248
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e
runc version: 2f7393a47307a16f8cee44a37b262e8b81021e3e
init version: 949e6fa
Security Options:
 apparmor
Kernel Version: 3.13.0-24-generic
Operating System: Ubuntu 14.04.5 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.784 GiB
Name: pocker
ID: NYDY:S5H7:JH2O:7W33:223L:TMPS:74VW:7WFX:XKDP:LPNG:PUFT:4WUU
Docker Root Dir: /data/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
WARNING: No kernel memory limit support
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by Puneeth-n** [comment]
@creynders The only workaround I found was to restart the docker daemon to remove the Dead container.

**Comment by cpuguy83** [comment]
This should be fixed completely in 17.06 by 54dcbab25ea4771da303fa95e0c26f2d39487b49
